client/landing/stepper/declarative-flow/internals/steps-repository/woo-confirm/index.tsx

```diff
@@ -46,68 +49,51 @@ const WooConfirm: Step = function WooCommerceConfirm( { navigation } ) {
 	const { __ } = useI18n();
 	const site = useSite();
 	const siteId = site && site?.ID;
-	const isAtomicSite = useSelect(
-		( select ) => siteId && ( select( SITE_STORE ) as SiteSelect ).isSiteAtomic( siteId ),
-		[ siteId ]
-	);
-	const { requestLatestAtomicTransfer } = useDispatch( SITE_STORE );
-	const stepProgress = useSelect(
-		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getStepProgress(),
-		[]
-	);
 
+	const { requestLatestAtomicTransfer } = useDispatch( SITE_STORE );
 	useEffect( () => {
 		if ( ! siteId ) {
 			return;
 		}
-
 		requestLatestAtomicTransfer( siteId );
 	}, [ requestLatestAtomicTransfer, siteId ] );
 
-	const eligibilityHolds = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getEligibilityHolds( siteId ),
-		[ siteId ]
-	);
-	const eligibilityWarnings = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getEligibilityWarnings( siteId ),
-		[ siteId ]
-	);
-	const wpcomSubdomainWarning = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getWpcomSubdomainWarning( siteId ),
-		[ siteId ]
-	);
-	const warnings = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getNonSubdomainWarnings( siteId ),
-		[ siteId ]
-	);
-	const latestAtomicTransfer = useSelect(
-		( select ) => ( select( SITE_STORE ) as SiteSelect ).getSiteLatestAtomicTransfer( siteId || 0 ),
-		[ siteId ]
-	);
-	const latestAtomicTransferError = useSelect(
-		( select ) =>
-			( select( SITE_STORE ) as SiteSelect ).getSiteLatestAtomicTransferError( siteId || 0 ),
+	const transferEligibility = useSelect(
+		( select ) => select( transferStore ).getAutomatedTransferEligibility( siteId ),
 		[ siteId ]
 	);
+
+	const { eligibilityHolds, eligibilityWarnings, wpcomSubdomainWarning, warnings } =
+		useMemo( () => {
+			return {
+				eligibilityHolds: getEligibilityHolds( transferEligibility ),
+				eligibilityWarnings: getEligibilityWarnings( transferEligibility ),
+				wpcomSubdomainWarning: getWpcomSubdomainWarning( transferEligibility ),
+				warnings: getNonSubdomainWarnings( transferEligibility ),
+			};
+		}, [ transferEligibility ] );
```

Comment:
Interesting call to pull those out. While it looks like a good optimization, I wonder if you've done any comparison between having those inside the `useSelect()` and outside of it in a `useMemo()`.

Reply:
That's a good point, I hadn't considered putting them inside the useSelect call. I agree that would have the same effect.